### ISSUE TITLE
--dry-run stats feature

### ISSUE THREAD

**Issue by Qwatuz** [rclone/rclone#1817]
As posted here in the forums: https://forum.rclone.org/t/dry-run-stats/4214

My feature request is for more information when running `sync`, `copy` or `move`  with `dry-run` to display the outcome of the command, e.g how many GBs of data will be synced to destination and how many GBs will be removed at destination.

This would be great to know how much data is remaining for a copy/sync. 

Currently the displayed information is:

```
Transferred:      0 Bytes (0 Bytes/s)
Errors:                 0
Checks:              5778
Transferred:         8179
Elapsed time:    37m21.9s
```

A neat feature could look like:

```
Transferred:      0 Bytes (0 Bytes/s)
Errors:                 0
Checks:              5778 (9,327.43 GB)
Deleted: 2,172 (182.27 GB)
Transferred:         8179 (12,294.23 GB)
Elapsed time:    37m21.9s
```

**Comment by ncw** [comment]
rsync tells you how much data it would transfer using `--dry-run`.

I like the idea of showing the data checked, deleted and transferred.

**Comment by smeijer** [comment]
Let's start with the deleted, I was surprised to found out that this information is missing in the output 😳

**Comment by Qwatuz** [comment]
Been a while since I posted this, was there enough interest in adding this feature @ncw ?

**Comment by ncw** [comment]
@Qwatuz wrote:
> Been a while since I posted this, was there enough interest in adding this feature

Not yet - want to have a go?

**Comment by Qwatuz** [comment]
> 
> 
> @Qwatuz wrote:
> 
> > Been a while since I posted this, was there enough interest in adding this feature
> 
> Not yet - want to have a go?

I wish I knew where to begin! Sorry I don't have a clue about this yet.

**Comment by SKnight79** [comment]
I was just coding around rsync to get these stats and I wanted to do the same for clone. Is there any way to know how much data would transferred.

**Comment by ncw** [comment]
Actually this is being discussed and implemented in https://github.com/rclone/rclone/issues/4624

**Comment by albertony** [comment]
Is this fixed by the mentioned PR, released in v1.54, and should be closed? Or is there a nuance, still missing?

Quick test:
```
C:\rclone>.\rclone sync docs C:\Temp\docs --dry-run
2021/09/07 13:00:44 NOTICE: content/bugs.md: Skipped copy as --dry-run is set (size 1.457Ki)
2021/09/07 13:00:44 NOTICE: content/compress.md: Skipped copy as --dry-run is set (size 4.384Ki)
2021/09/07 13:00:44 NOTICE: content/contact.md: Skipped copy as --dry-run is set (size 736)
2021/09/07 13:00:44 NOTICE: content/cache.md: Skipped copy as --dry-run is set (size 21.188Ki)
2021/09/07 13:00:44 NOTICE: content/changelog.md: Skipped copy as --dry-run is set (size 182.359Ki)
2021/09/07 13:00:44 NOTICE: content/chunker.md: Skipped copy as --dry-run is set (size 18.052Ki)
2021/09/07 13:00:44 NOTICE: content/crypt.md: Skipped copy as --dry-run is set (size 25.030Ki)
2021/09/07 13:00:44 NOTICE: content/downloads.md: Skipped copy as --dry-run is set (size 5.574Ki)
2021/09/07 13:00:44 NOTICE: content/docker.md: Skipped copy as --dry-run is set (size 21.726Ki)
2021/09/07 13:00:44 NOTICE: content/docs.md: Skipped copy as --dry-run is set (size 84.401Ki)
2021/09/07 13:00:44 NOTICE: content/faq.md: Skipped copy as --dry-run is set (size 8.298Ki)
2021/09/07 13:00:44 NOTICE: content/donate.md: Skipped copy as --dry-run is set (size 2.292Ki)
2021/09/07 13:00:44 NOTICE: content/drive.md: Skipped copy as --dry-run is set (size 45.304Ki)
2021/09/07 13:00:44 NOTICE: content/dropbox.md: Skipped copy as --dry-run is set (size 14.307Ki)
2021/09/07 13:00:44 NOTICE: content/fichier.md: Skipped copy as --dry-run is set (size 5.145Ki)
2021/09/07 13:00:44 NOTICE: content/filefabric.md: Skipped copy as --dry-run is set (size 7.175Ki)
2021/09/07 13:00:44 NOTICE: content/flags - Copy.md: Skipped delete as --dry-run is set (size 64.630Ki)
2021/09/07 13:00:44 NOTICE: content/gui - Copy.md: Skipped delete as --dry-run is set (size 3.541Ki)
2021/09/07 13:00:44 NOTICE: content/hdfs - Copy.md: Skipped delete as --dry-run is set (size 5.932Ki)
2021/09/07 13:00:44 NOTICE: content/filtering - Copy.md: Skipped delete as --dry-run is set (size 24.162Ki)
2021/09/07 13:00:44 NOTICE: content/ftp - Copy.md: Skipped delete as --dry-run is set (size 8.275Ki)
2021/09/07 13:00:44 NOTICE: content/googlecloudstorage - Copy.md: Skipped delete as --dry-run is set (size 16.750Ki)
2021/09/07 13:00:44 NOTICE: content/googlephotos - Copy.md: Skipped delete as --dry-run is set (size 13.390Ki)
2021/09/07 13:00:44 NOTICE:
Transferred:      447.410 KiB / 447.410 KiB, 100%, 0 B/s, ETA -
Checks:               186 / 186, 100%
Deleted:                7 (files), 0 (dirs)
Transferred:           16 / 16, 100%
Elapsed time:         1.5s
```

**Comment by ncw** [comment]
I think this is now implemented and if there is any improvements we need then it should be in a new issue, so I'll close this one - thanks @albertony
